File: javascript/7kyu/Sum of numbers from 0 to N.js
// https://www.codewars.com/kata/56e9e4f516bcaa8d4f001763

// We want to generate a function that computes the series starting from 0 and ending until the given number.

var SequenceSum = (function() {
    function SequenceSum() {}
  
    SequenceSum.showSequence = function(count) {
      var sum = 0;
      var str = "";
      if (count === 0) {
        return "0=0";
      } else if (count < 0) {
        return count + "<0";
      } else {
        for (var i = 0; i < count; i++) {
          sum += i;
          str += i + "+";
        };
      sum += count;
      str += count + " = " + sum;
      return str;
      };
    };
  
    return SequenceSum;
  
  })();

File: java/7kyu/Sum of numbers from 0 to N.java
// https://www.codewars.com/kata/56e9e4f516bcaa8d4f001763

// We want to generate a function that computes the series starting from 0 and ending until the given number.

public class SequenceSum{

  public static String showSequence(int value){
    String ans = "";
    if (value < 0) return "" + value + "<0";
    if (value == 0) return "0=0";
    for (int i = 0; i <= value; i++) ans += "" + i + "+";
    return ans.substring(0,ans.length() - 1) + " = " + (value * (value + 1) /2);
  }
}